pkg/kosmosctl/join/join.go

```diff
@@ -0,0 +1,207 @@
+package join
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	k8syaml "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctlutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/kosmos.io/kosmos/pkg/kosmosctl/manifest"
+	"github.com/kosmos.io/kosmos/pkg/kosmosctl/util"
+	"github.com/kosmos.io/kosmos/pkg/utils"
+	"github.com/kosmos.io/kosmos/pkg/version"
+)
+
+var joinExample = templates.Examples(i18n.T(`
+		# join member1-cluster to master control plane
+		kosmosctl join --cluster-kubeconfig=[member-kubeconfig] -f member1-cluster.yaml --master-kubeconfig=[master-kubeconfig]
+
+		# join member1-cluster to current master control plane
+		kosmosctl join --cluster-kubeconfig=[member-kubeconfig] -f member1-cluster.yaml
+`))
+
+type CommandJoinOptions struct {
+	File              string
+	MasterKubeConfig  string
+	ClusterKubeConfig string
+
+	DynamicClient *dynamic.DynamicClient
+	Client        kubernetes.Interface
+}
+
+// NewCmdJoin join this to Clusterlink control plane
+func NewCmdJoin(f ctlutil.Factory) *cobra.Command {
+	o := &CommandJoinOptions{}
+
+	cmd := &cobra.Command{
+		Use:                   "join",
+		Short:                 i18n.T("Join this to clusterlink control plane"),
```

Comment:
what 18n means? seems so strange

Reply:
Quoted from kubectl, internationalization/i18n？the same as kubernetes/k8s

Reply:
18n is an abbreviation for internationalization, and it shows different languages depending on locale, as I've seen used in other codes